### ISSUE TITLE
[Bug] Linux/BlueZ: Application hangs for 15s during characteristic cleanup on abrupt device disconnect

### ISSUE THREAD

**Issue by rk4248** [simpleble/simpleble#482]
Environment:
OS: Linux (Raspberry Pi / BlueZ stack)
Library Component: backends/linux/PeripheralLinux.cpp
Protocol: Bluetooth Low Energy (GATT Server)

Description of the Issue:
A severe blocking issue occurs in the Linux backend when a connected BLE peripheral (with active notifications) abruptly loses power or goes out of range.

During the cleanup phase—either triggered by an explicit disconnect() call or when the Peripheral object is being destroyed—the library iterates over the services and calls bluez_characteristic->stop_notify() synchronously (around line 372 in PeripheralLinux.cpp).

Because the physical device is dead, BlueZ cannot successfully send the 0x0000 value to the CCCD. Consequently, the underlying D-Bus call blocks the main thread for approximately 15 seconds until the D-Bus/Supervision timeout is reached. The library eventually catches the exception: Exception during characteristic cleanup: D-Bus call timed out, but the 15-second main thread freeze makes it unsuitable for real-time gateway applications handling multiple devices.

Steps to Reproduce:

Connect to any BLE Peripheral using SimpleBLE on a Linux machine.
Subscribe to at least one characteristic using notify().
Abruptly cut the physical power to the Peripheral device (simulating link loss).
Attempt to call disconnect() or allow the Peripheral object to go out of scope.
Result: The executing thread hangs completely for ~15 seconds before resuming.

Proposed Solution (Asynchronous Cleanup):
To maintain the "graceful disconnect" feature for healthy devices while preventing the D-Bus timeout from paralyzing the main application thread, the stop_notify() call should be delegated to a detached background thread.

Suggested Code Change (PeripheralLinux.cpp):

Current Implementation (Blocking):

C++
try {
    if (bluez_characteristic->notifying()) {
        bluez_characteristic->stop_notify();
    }
} catch (std::exception const& e) {
    SIMPLEBLE_LOG_WARN(fmt::format("Exception during characteristic cleanup: {}", e.what()));
}

Proposed Fix (Non-blocking Fire-and-Forget):

C++
try {
    if (bluez_characteristic->notifying()) {
        auto char_ptr = bluez_characteristic; // Copy the shared_ptr to keep D-Bus object alive
        std::thread([char_ptr]() {
            try {
                char_ptr->stop_notify();
            } catch (...) {
                // Ignore D-Bus timeout exceptions in the background thread
            }
        }).detach();
    }
} catch (std::exception const& e) {
    SIMPLEBLE_LOG_WARN(fmt::format("Exception during characteristic wrapper cleanup: {}", e.what()));
}
Impact:
Implementing this change ensures that the main application thread returns immediately, while the D-Bus timeout is safely handled in the background, significantly improving the robustness of the Linux backend in unstable RF environments.

**Comment by kdewald** [comment]
Thanks for the detailed report. This should be fixed on `main` now.

The cleanup path now avoids calling `StopNotify` from the passive disconnect callback after BlueZ has already reported the device as gone, while still stopping active notifications during an explicit disconnect when the device is still considered connected.

Closing this, but please reopen or comment if you can still reproduce the delay with the latest `main`.

-- Riley
